### PR TITLE
Do not scroll for UIAlertControllers

### LINKED ISSUE #14041: [iOS] Page Contents shift down when using Prompt with input field.
### Description

https://user-images.githubusercontent.com/898335/226103670-98c0861b-b891-428d-adc4-7c05964f26b9.mov

I believe this is caused by [this PR](https://github.com/dotnet/maui/commit/16fcd7e481c1b883aca9fe3533aebd445e212ead). The inner contents of the page are shifting down to match the keyboard height, but I am using a standard iOS prompt, so there's no reason for it to shift.

### Steps to Reproduce

1. Create an alert with a input field.
2. Watch as the inner contents shift.

### Link to public reproduction project repository

https://github.com/dotnet/maui/tree/main/src/Controls/samples/Controls.Sample

### Version with bug

Unknown/Other (please specify)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

All iOS Versions

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by tj-devel709**
### Description of Change

When a UIAlertController is rendered, the scrolling is already handled behind the scenes. There is not a need for any additional scrolling to happen here so we can ignore the scrolling on the KeyboardAutoManagerScroll side.

https://user-images.githubusercontent.com/50846373/226692800-e10f2d34-f2ce-4a42-8e73-f42449d53925.mov

### Issues Fixed

Fixes #14041

### CODE REVIEW

**Review by drasticactions — APPROVED** [review]
Tried on the PR branch and it worked for me.

**Review by PureWeen — CHANGES_REQUESTED** [review]
Looks good! Add tests and toggle the `IsKeyboardScrolling` flag!

File: src/Controls/tests/DeviceTests/Elements/Entry/EntryTests.iOS.cs [hunk 1]
```diff
@@ -36,5 +40,57 @@ int GetPlatformSelectionLength(EntryHandler entryHandler)
...
+					if (win is UIWindow uIWindow)
+					{
+						var alert = UIAlertController.Create("Popup Alert", "This is a popup", UIAlertControllerStyle.Alert);
+
+						alert.AddTextField((textfield) =>
+						{
+							textfield.Placeholder = "Placeholder Text";
+						});
+
+						_ = uIWindow.RootViewController.PresentViewControllerAsync(alert, true);
```
Comment (PureWeen) [inline src/Controls/tests/DeviceTests/Elements/Entry/EntryTests.iOS.cs:82]: I don't think we want to ignore past these Async calls. 

My thoughts for this test
1) await these calls
2) make this test part of a Collection. You can see where this is done a few places with an inner class

[Category(TestCategory.Accessibility)]
		[Collection(ControlsHandlerTestBase.RunInNewWindowCollection)]
		public class InNewWindowCollection : ControlsHandlerTestBase
		{

3) use entry.GetBoundingBox to validate that the location of the entry hasn't shifted

Comment (PureWeen) [reply src/Controls/tests/DeviceTests/Elements/Entry/EntryTests.iOS.cs:82]: I don't think we want to ignore past these Async calls. 

My thoughts for this test
1) await these calls
2) make this test part of a Collection. You can see where this is done a few places with an inner class

[Category(TestCategory.Accessibility)]
		[Collection(ControlsHandlerTestBase.RunInNewWindowCollection)]
		public class InNewWindowCollection : ControlsHandlerTestBase
		{

3) use entry.GetBoundingBox to validate that the location of the entry hasn't shifted

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/tests/DeviceTests/Elements/Entry/EntryTests.iOS.cs [hunk 2]
```diff
@@ -43,115 +43,181 @@ int GetPlatformSelectionLength(EntryHandler entryHandler)
...
-					cell.SetBinding(EntryCell.TextProperty, ".");
-					return cell;
-				}),
-				ItemsSource = Enumerable.Range(0, 10).Select(i => $"EntryCell {i}").ToList()
-			};
+					var contentViewHandler = CreateHandler<LayoutHandler>(layout);
+					await contentViewHandler.PlatformView.AttachAndRun(async () =>
+					{
+						var entryBoxFunc = entry.GetBoundingBox;
+						var initialEntryBox = entryBoxFunc.Invoke();
```
Comment (PureWeen) [inline src/Controls/tests/DeviceTests/Elements/Entry/EntryTests.iOS.cs:77]: Remove invoke calls

**Review by PureWeen — CHANGES_REQUESTED** [review]

**Review by PureWeen — APPROVED** [review]

**Pull request merged**